src/app/components/task-progress/task-progress.component.ts

```diff
@@ -0,0 +1,30 @@
+import { Component, Input, inject } from '@angular/core';
+import { TaskService } from '../../service/task.service';
+import { Task } from '../../model/task/task';
+
+@Component({
+  selector: 'app-task-progress',
+  standalone: true,
+  imports: [],
+  templateUrl: './task-progress.component.html',
+  styleUrl: './task-progress.component.scss'
+})
+export class TaskProgressComponent {
+  private taskService = inject(TaskService);
+  taskTotal : number = 0;
+  taskAssigned : number = 0;
+  taskCompleted : number = 0;
+
+  ngOnInit(){
+    this.taskTotal = this.taskService.getTasks().length;
+    this.taskCompleted = this.taskService.getTasks().filter(t => t.isCompleted()).length;
+    this.taskAssigned = this.taskTotal - this.taskCompleted;
```

Comment:
could you refactor those lines in a method that you can call inside your onInit and during the subscribe of your updatedTask